service/src/main/java/bio/terra/cbas/config/AsyncConfiguration.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.cbas.config;
+
+import bio.terra.cbas.controllers.util.AsyncExceptionHandler;
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dao.RunSetDao;
+import java.util.concurrent.ThreadPoolExecutor;
+import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.bind.ConstructorBinding;
+import org.springframework.context.annotation.Bean;
+import org.springframework.scheduling.annotation.AsyncConfigurer;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+@EnableAsync
+@ConfigurationProperties(prefix = "cbas.async.submission")
+public class AsyncConfiguration implements AsyncConfigurer {
+  private final int coreThreadPoolSize;
+  private final int maxThreadPoolSize;
+  private final int queueCapacity;
+
+  @Autowired private final RunDao runDao;
+  @Autowired private final RunSetDao runSetDao;
```

Comment:
Question for reviewers: Is this the right way to auto-wired DAO references in a class that also has annotation `ConfigurationProperties`? I had tried adding `Component` annotation and it gave below error. I tried removing `ConstructorBinding` but it still gave the same error. We already have `ConfigurationPropertiesScan` enabled in [App.java](https://github.com/DataBiosphere/cbas/blob/c328335c187693607a408d63faf791aee16305cf/service/src/main/java/bio/terra/cbas/App.java#L34)
```
Description:
AsyncConfiguration is annotated with @ConstructorBinding but it is defined as a regular bean which caused dependency injection to fail.

Action:
Update your configuration so that AsyncConfiguration is defined via @ConfigurationPropertiesScan or @EnableConfigurationProperties.
```

Reply:
I removed the `@Autowired` and `@ConstructorBinding` annotation on the constructor and it gave a run time error
```
Exception handler for async method 'public void bio.terra.cbas.service.RunSetsService.triggerWorkflowSubmission(...)' threw unexpected exception itself
java.lang.NullPointerException: Cannot invoke "bio.terra.cbas.dao.RunDao.getRuns(bio.terra.cbas.dao.RunDao$RunsFilters)" because "this.runDao" is null
```
Upon addition of `@Component` it threw the same error as mentioned in first thread `...defined as a regular bean which caused dependency injection to fail`